### PR TITLE
[extension/pprof] Improve documentation for pprof usage

### PR DESCRIPTION
**Pull request by Jamie-Ullerich**
**Description:** Update readme with instructions on how to use pprof

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/26095

**Testing:** Not applicable since no functional code was changed.

**Documentation:** The extension readme now contains information what a profile is, how it can be generated and analysed.

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
Please sign the CLA

File: extension/pprofextension/README.md [hunk 1]
```diff
@@ -48,3 +48,60 @@ extensions:
...
+There are different endpoints for other types of profiles.
+For instance, the memory can be analyzed using `go tool pprof http://localhost:1777/debug/pprof/heap`.
+To see all available profiles, visit `http://localhost:1777/debug/pprof/` in your browser.
+
+#### Analyzing a profile
+
+After running the above command to save the profile, _pprof_ will enter the interactive mode.
+From here, the profiles can be analyzed.
+
+Use the command `web` to open an image of the complete call graph in your browser.
```
Comment (songy23) [inline extension/pprofextension/README.md:80]: May want to also mention [Graphviz](https://graphviz.org/) as a prerequisite for `web`

**Review by songy23 — APPROVED** [review]
Minor suggestion, otherwise LGTM

**Review by TylerHelmuth — APPROVED** [review]
Looks good

**Review by atoulme — APPROVED** [review]

**Pull request merged**